include/cudaq/Optimizer/Dialect/Quake/QuakeInterfaces.td

```diff
@@ -108,6 +108,16 @@ def MeasurementInterface : OpInterface<"MeasurementInterface"> {
           return {attr.getValue().str()};
         return {};
       }]
+    >,
+    InterfaceMethod<
+      /*desc=*/ "Set the registerName attribute",
+      /*retType=*/ "void",
+      /*methodName=*/ "setRegisterName",
+      /*args=*/ (ins "llvm::StringRef":$name),
+      /*methodBody=*/ [{
+        $_op->setAttr("registerName",
+                      mlir::OpBuilder($_op).getStringAttr(name));
```

Comment:
Looks like I made it harder than it needed to be then 😄 . I'll change it to your suggestion. Thanks.

Reply:
I think it is slightly more conventional to call ctors directly rather than build an `OpBuilder` on-the-fly. (Although, I could be swayed by actual evidence. heh) At any rate, that'd be something like
```
mlir::StringAttr::get($_op.getContext(), name)
```